substrate/frame/contracts/src/wasm/runtime.rs

```diff
@@ -2136,26 +2122,27 @@ pub mod env {
 		ctx.charge_gas(RuntimeCosts::CopyFromContract(msg_len))?;
 		let message: VersionedXcm<CallOf<E::T>> =
 			ctx.read_sandbox_memory_as_unbounded(memory, msg_ptr, msg_len)?;
+		ensure_executable::<E::T>(&message)?;
 
 		let execute_weight =
 			<<E::T as Config>::Xcm as ExecuteController<_, _>>::WeightInfo::execute();
 		let weight = ctx.ext.gas_meter().gas_left().max(execute_weight);
```

Comment:
Why would we even pass in all the remaining weight if it is bigger than `execute_weight`? `execute_weight` should be  the upper bound for the weight than can be used.  We are not doing that either for `call_runtime`. This seems to be the root of the problem.

Why not just use the `execute_weight`?

Reply:
execute_weight is actually the base weight for executing the pallet_xcm dispatchable, then you top up with max_weight to pay for it.

So, I think my reasoning here was that we can't pass the function anything less than execute_weight. 

https://github.com/paritytech/polkadot-sdk/blob/a8ea8610c15a2426ea20067b52fb106ee6dc97f8/polkadot/xcm/pallet-xcm/src/lib.rs#L1021-L1036

Reply:
Ahh yes you are right. I thought this was the upper limit from a `Weigher`.